packages/eddsa-pcd/src/EdDSAPCD.ts

```diff
@@ -146,7 +146,9 @@ export async function serialize(
 }
 
 export async function deserialize(serialized: string): Promise<EdDSAPCD> {
-  return JSON.parse(serialized, reviver);
+  const { id, claim, proof } = JSON.parse(serialized, reviver) as EdDSAPCD;
```

Comment:
I think casting to the concrete PCD type here is a bit misleading, and also confusing for readers. I could imagine someone wondering - "if it's a `EdDSAPCD`, why do I need to create a new instance of it?".

We should also throw an `Error` if any of `id`, `claim`, or `proof` are `undefined`.

same comment for the rest of the PCDs.

Reply:
I'll do it!

Question, why are the `serialize` and `deserialize` functions async?

Reply:
This still feels a bit vulnerable to me. A malicious feed or redirect URL could include a "serialized PCD" that contains arbitrary data in the `claim`, `id` or `proof` fields. It won't be `undefined` but it will still be invalid.

The pattern now is "parse it from JSON, check it exists, then pass it to a constructor which just assigns it to a member variable". The variable then gets passed to other functions without ever being checked for validity, e.g.
```
  const publicSignals = publicSignalsFromClaim(pcd.claim);
  return groth16.verify(vkey, publicSignals, pcd.proof);
```
I don't know that passing malicious inputs to these functions can cause harm, but we could eliminate the possibility by validating these inputs more rigorously.

Reply:
@robknight - I agree that we should be careful about malicious inputs. i can't think of an example of a malicious input though. I think we should address that in another pull request attached to another issue - https://github.com/proofcarryingdata/zupass/issues/716